tree/ntuple/v7/inc/ROOT/RPageStorage.hxx

```diff
@@ -276,6 +277,53 @@ mapped into memory. The page source also gives access to the ntuple's meta-data.
 */
 // clang-format on
 class RPageSource : public RPageStorage {
+public:
+   /// An RAII wrapper used for the read-only access to RPageSource::fDescriptor
+   class RSharedDescriptorGuard {
+      const RNTupleDescriptor &fDescriptor;
+      std::shared_mutex &fLock;
```

Comment:
`std::shared_mutex` is neither read nor write recursive: "[Within one thread, only one lock (shared or exclusive) can be acquired at the same time. ".](https://en.cppreference.com/w/cpp/thread/shared_mutex)   

I recommend that we stress this fact in the docs (failure will lead to dead-lock, I think), especially since it can happen indirectly.  For example, this "artifiicial" example (modifying fatally a bit code below) lead to a dead-lock:

```
   R__ASSERT(fieldId != kInvalidDescriptorId);
   auto guard =  GetSharedDescriptorGuard()->FindColumnId(fieldId, column.GetIndex());
   auto columnId = guard->FindColumnId(fieldId, column.GetIndex());
   R__ASSERT(columnId != kInvalidDescriptorId);
   
   R__ASSERT(GetNEntries() > 0);   // Fatal in debug build ... 

   fActiveColumns.emplace(columnId);
   return ColumnHandle_t{columnId, &column};
```

Reply:
Yep, done [here|https://github.com/root-project/root/pull/10185/commits/cb8cc487b10e211da3ac8ef3eca4ff7a765c7d86]